docs/architecture/decisions/0010-sdk-packaging.md

```diff
@@ -0,0 +1,55 @@
+# 0010. Create packages for CarbonAware SDK
+
+## Status
+
+Proposed
+
+## Context
+
+Current GSF CarbonAware software can't be consumed as [Nuget](https://www.nuget.org) Packages in case there is an application that requires to integrate with it. There are scenerarios where the current runtimes (CLI/WebApi) environments are not allowed to be used since all the functionality requires to be bundled together.
+The goal of this proposal is to show that the current GSF repository can provide a mechanism to generate `Nuget` packages that can be consumed by any application that wants to integrate with `GSF CarbonAware`.
+
+## Decision
+
+Having nuget packages available to be consumed by 3rd party application would help to integrate easily and it would open the possibilities to extend the SDK more.
+Currently there are seven (7) `dotnet` projects that can be set to be packaged with tools like `dotnet pack` or `msbuild -t:pack`.
+
+|CarbonAware Packages|
+|--------------------|
+|CarbonAware|
+|CarbonAware.Aggregators|
+|CarbonAware.DataSources.Json|
+|CarbonAware.DataSources.Registration|
+|CarbonAware.DataSources.WattTime|
+|CarbonAware.LocationSources|
+|CarbonAware.Tools.WattTimeClient|
+
+
+**[Must Address]** Creation of a minimum set of packages that can be integrate with a 3rd party application.
+  * Continues enhancing the SDK to support integration with different type of applications.
+  * Enhance project files (csproj) to incorporate package metadata.
+  * Enhance project files (csproj) to include resources data files.
+
+**[Should Address]** Creation of a package repository(ies).
+  * Use a local feed, private feed or nuget.org repositories.
```

Comment:
I think it is reasonable to ship packages from GitHub Packages as discussion at https://github.com/Green-Software-Foundation/carbon-aware-sdk/discussions/46 .

Reply:
We should think about other languages in the future - NuGet is for .Net only.

Reply:
I feel creating a nuget package would be great so we should discuss this ADR.  Whether it gets published is something we will have to discuss with the GSF due to maintenance overhead and commitments.